star_tides/api/controllers/guide_controller.py

```diff
@@ -0,0 +1,63 @@
+'''star_tides.api.controllers.guide_controller
+'''
+
+from star_tides.exceptions import InvalidParamError
+from star_tides.api.controllers.base_controller import Controller
+from star_tides.api.controllers import snake_case_dict
+from star_tides.core.actions.guide.create_guide_action import CreateGuideAction
+from star_tides.core.actions.guide.list_guide_action import ListGuidesAction
+from star_tides.core.actions.guide.get_guide_action import GetGuideAction
+from star_tides.core.actions.guide.delete_guide_action import DeleteGuideAction
+from star_tides.core.actions.guide.update_guide_action import UpdateGuideAction
+
+
+class ListGuidesController(Controller):
+    def process_request(self) -> dict:
+        args = Controller.get_query_params()
+        params = {}
+        if args.get('limit'):
+            params['limit'] = args.get('limit')
+        if args.get('offset'):
```

Comment:
Prefer to use page tokens instead of offsets since they are more flexible. A page token can just be an encoded offset for example.

Reply:
I'm not sure what page tokens are. Can you explain?

Reply:
They are basically opaque strings that can encode arbitrary information. The contract is that a serve passes back a response that contains the list of things and a page token for the next page. The client makes the same request but uses the new page token. This process iterates until the client has consumed all of the pages and the server responds with an empty page token. https://google.aip.dev/158 might be helpful here.

If you have stuff working with an offset, you could always just do something like: `page_token = base64encode(offset)`. The server can know how to encode/decode page tokens, but to the client, they are opaque.